
Avoid a buffer overrun. This does not expose us to a DoS since the kernel will
have automatically closed the overflow file descriptors on the receiving side.

Reported upstream at

	https://gitlab.freedesktop.org/dbus/dbus/-/issues/304

diff -wpruN '--exclude=*.orig' a~/dbus/dbus-sysdeps-unix.c a/dbus/dbus-sysdeps-unix.c
--- a~/dbus/dbus-sysdeps-unix.c	1970-01-01 00:00:00
+++ a/dbus/dbus-sysdeps-unix.c	1970-01-01 00:00:00
@@ -461,6 +461,7 @@ _dbus_read_socket_with_unix_fds (DBusSoc
                  * truncate. */
                 fds_to_use = (size_t) *n_fds;
 
+#ifndef __sun__
                 /* Close the excess fds to avoid DoS: if they stayed open,
                  * someone could send us an extra fd per message
                  * and we'd eventually run out. */
@@ -468,6 +469,7 @@ _dbus_read_socket_with_unix_fds (DBusSoc
                   {
                     close (payload[i]);
                   }
+#endif
               }
 
             memcpy (fds, payload, fds_to_use * sizeof (int));
